pkg/api/errors/errors.go

```diff
@@ -297,6 +297,17 @@ func NewGenericServerResponse(code int, verb, kind, name, serverMessage string,
 	case len(kind) > 0:
 		message = fmt.Sprintf("%s (%s %s)", message, strings.ToLower(verb), kind)
 	}
+	var causes []api.StatusCause
+	if isUnexpectedResponse {
+		causes = []api.StatusCause{
+			{
+				Type:    api.CauseTypeUnexpectedServerResponse,
+				Message: serverMessage,
+			},
+		}
+	} else {
+		causes = nil
```

Comment:
How about PathInvalid as cause for 404s from go-restful and DoesNotExist for 404s from etcd (i.e., the cause reported by NewNotFound)?